engine/flink/executor/src/test/scala/pl/touk/nussknacker/engine/process/compiler/ProcessNameValidationSpec.scala

```diff
@@ -0,0 +1,65 @@
+import pl.touk.nussknacker.engine.api.SpelExpressionExcludeList
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.ProcessNameValidationError
+import pl.touk.nussknacker.engine.api.process.{ClassExtractionSettings, LanguageConfiguration}
+import pl.touk.nussknacker.engine.api.typed.typing.Typed
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.compile.ProcessValidator
+import pl.touk.nussknacker.engine.definition.DefinitionExtractor.ObjectDefinition
+import pl.touk.nussknacker.engine.definition.ProcessDefinitionExtractor.{ExpressionDefinition, ProcessDefinition}
+import pl.touk.nussknacker.engine.dict.SimpleDictRegistry
+import pl.touk.nussknacker.engine.process.ProcessNameValidatorFactory
+import pl.touk.nussknacker.engine.process.helpers.ProcessTestHelpers
+import pl.touk.nussknacker.engine.process.helpers.SampleNodes.SimpleRecord
+import pl.touk.nussknacker.engine.testing.ProcessDefinitionBuilder
+import pl.touk.nussknacker.engine.{CustomProcessValidatorFactory, CustomProcessValidatorLoader}
+import pl.touk.nussknacker.test.ClassLoaderWithServices
+
+class ProcessNameValidationSpec extends AnyFunSuite with Matchers with ProcessTestHelpers with Inside {
+
+  private val processDefinition = ProcessDefinition[ObjectDefinition](
+    services = Map(),
+    sourceFactories = Map("source" -> ObjectDefinition(List.empty, Typed[SimpleRecord])),
+    sinkFactories = Map("sink" -> ObjectDefinition.noParam),
+    customStreamTransformers = Map(),
+    signalsWithTransformers = Map(),
+    expressionConfig = ExpressionDefinition(Map.empty, List.empty, List.empty, LanguageConfiguration.default, optimizeCompilation = false, strictTypeChecking = true,
+      Map.empty, hideMetaVariable = false, strictMethodsChecking = true, staticMethodInvocationsChecking = false,
+      methodExecutionForUnknownAllowed = false, dynamicPropertyAccessAllowed = false, spelExpressionExcludeList = SpelExpressionExcludeList.default,
+      customConversionsProviders = List.empty),
+    settings = ClassExtractionSettings.Default
+  )
+
+  test("we are checking name validation") {
+
+    val processWithInvalidName =
+      ScenarioBuilder
+        .streaming("invalid+name")
+        .source("sourceId", "source")
+        .emptySink("emptySink", "sink")
+        .toCanonicalProcess
+
+    val customProcessValidator = ClassLoaderWithServices.withCustomServices(List((classOf[CustomProcessValidatorFactory], classOf[ProcessNameValidatorFactory]))) { loader =>
+      CustomProcessValidatorLoader.loadProcessValidators(loader, ConfigFactory.empty())
+    }
+
+    val processValidator = ProcessValidator.default(
+      ProcessDefinitionBuilder.withEmptyObjects(processDefinition),
+      new SimpleDictRegistry(Map.empty),
+      customProcessValidator)
+
+    val validationResult = processValidator
+      .validate(processWithInvalidName)
+
+    validationResult.result shouldBe invalidNel(
```

Comment:
This check will give more agility to further changes of messages if you don't validate message but only class of exception. I mean:
```
validationResult.result should matchPattern {
  case ProcessNameValidationError("invalid+name", _, _) => _
}
```